### ISSUE TITLE
Address indexing does not gracefully shutdown

### ISSUE THREAD

**Issue by carryforward** [btcsuite/btcd#448]
I am running code tagged 0.11.1.  The blockchain is fully synced.  The address index had been dropped, due to upgrading from v.10.  

When I type ctrl+c btcd starts to shut down.  It doesn't look like the address indexing subsystem gets this shutdown message.  It continues to try to index blocks after shutdown starts.
```
15:56:58 2015-06-15 [INF] ADXR: Indexed addresses of 1 block in the last 30.9s (1369 transactions, height 312115, 2014-07-23 07:06:19 -0500 CDT)
15:57:10 2015-06-15 [INF] ADXR: Indexed addresses of 4 blocks in the last 12.35s (3045 transactions, height 312119, 2014-07-23 08:15:59 -0500 CDT)
15:57:21 2015-06-15 [INF] ADXR: Indexed addresses of 5 blocks in the last 10.11s (3255 transactions, height 312124, 2014-07-23 09:18:41 -0500 CDT)
15:57:42 2015-06-15 [INF] ADXR: Indexed addresses of 4 blocks in the last 21.61s (2133 transactions, height 312128, 2014-07-23 10:00:02 -0500 CDT)
15:57:55 2015-06-15 [INF] ADXR: Indexed addresses of 8 blocks in the last 12.95s (3497 transactions, height 312136, 2014-07-23 11:14:03 -0500 CDT)
15:58:45 2015-06-15 [INF] ADXR: Indexed addresses of 2 blocks in the last 50.16s (1675 transactions, height 312138, 2014-07-23 11:42:17 -0500 CDT)
15:58:56 2015-06-15 [INF] ADXR: Indexed addresses of 15 blocks in the last 10.55s (8085 transactions, height 312153, 2014-07-23 13:59:52 -0500 CDT)
15:59:07 2015-06-15 [INF] ADXR: Indexed addresses of 6 blocks in the last 11.55s (3105 transactions, height 312159, 2014-07-23 15:01:56 -0500 CDT)
^C15:59:18 2015-06-15 [INF] BTCD: Received SIGINT (Ctrl+C).  Shutting down...
15:59:18 2015-06-15 [INF] BTCD: Gracefully shutting down the server...
15:59:18 2015-06-15 [WRN] SRVR: Server shutting down
15:59:18 2015-06-15 [WRN] RPCS: RPC server shutting down
15:59:18 2015-06-15 [INF] RPCS: RPC server shutdown complete
15:59:18 2015-06-15 [INF] ADXR: Address indexer shutting down
15:59:18 2015-06-15 [INF] BMGR: Block manager shutting down
15:59:18 2015-06-15 [INF] AMGR: Address manager shutting down
15:59:18 2015-06-15 [INF] SRVR: Server shutdown complete
15:59:18 2015-06-15 [INF] BTCD: Gracefully shutting down the database...
15:59:18 2015-06-15 [INF] BTCD: Shutdown complete
15:59:19 2015-06-15 [ERR] ADXR: Unable to index transactions of block: transaction 0d2771bf47e80938e3d8233a3e74eea3012fbc65e7846a0643892debddd19af9 not found
15:59:19 2015-06-15 [ERR] ADXR: Unable to index transactions of block: transaction 90e8e57875bc14ead5c8c59abd23e87ad69e5da912efaa366d216ac4bb5194bf not found
```

**Comment by davecgh** [comment]
Thanks for the report.

I'll look at this a little closer, but I suspect it's just a matter of the fact the indexer goroutine is asynchronous the items were already in the processing queue when the shutdown was received, and they aren't being removed/cancelled on receiving the shutdown signal.

I believe that since the database is closed and the indexing is failing as expected, it shouldn't be causing any real issues, but it should definitely be cleaned up to cancel/discard.

**Comment by davecgh** [comment]
@Dirbaio.  Since you're currently reworking the address index against the new database code, would you make sure this is resolved along the way?

**Comment by Dirbaio** [comment]
Sure, I will.

**Comment by davecgh** [comment]
This was fixed by a combination of #693 and #733
